jetstream/cli.py

```diff
@@ -56,28 +54,71 @@ class AllType:
 
 class ExecutorStrategy(Protocol):
     project_id: str
-    dataset_id: str
 
-    def __init__(self, project_id: str, dataset_id: str, *args, **kwargs) -> None:
+    def __init__(self, project_id: str, *args, **kwargs) -> None:
         ...
 
-    def execute(self, worklist: Iterable[Tuple[str, AnalysisSpec, datetime]]) -> bool:
+    def execute(
+        self,
+        worklist: Iterable[Tuple[str, datetime]],
+        configuration_map: Mapping[str, TextIO] = {},
+    ) -> bool:
         ...
 
 
+@attr.s(auto_attribs=True)
+class ArgoExecutorStrategy:
+    project_id: str
+    zone: str
+    cluster_id: str
+    monitor_status: bool
+    experiment_getter: Callable[[], ExperimentCollection] = ExperimentCollection.from_experimenter
+
+    WORKLFOW_DIR = Path(__file__).parent / "workflows"
+    RUN_WORKFLOW = WORKLFOW_DIR / "run.yaml"
+
+    def execute(self, worklist, _configuration_map: Mapping[str, TextIO] = {}):
```

Comment:
I think this should not ignore a configuration_map if a non-empty map is provided; it would be nice to support that but something should at least error out.

Also I think the name of the argument is part of the API contract so it shouldn't be underscored, but if mypy isn't complaining then ¯\\\_(ツ)\_/¯